### ISSUE TITLE
poetry installs latest keyring but requires incompatible importlib-metadata

### ISSUE THREAD

**Issue by jclerman** [python-poetry/poetry#5226]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: CircleCI docker image: cimg/python:3.7.10 
- **Poetry version**: 1.1.13
- **Python version**: 3.7.10

## Issue
Immediately after installing the latest version of poetry (1.1.13) via `pip` (i.e., `pip install -U poetry`), a call to `poetry config http-basic.myrepository $USERNAME $PASSWORD` fails with the following:
```
  TypeError

  entry_points() got an unexpected keyword argument 'group'

  at ~/.pyenv/versions/3.7.10/lib/python3.7/site-packages/keyring/backend.py:196 in _load_plugins
      192│     of the plugin at initialization time.
      193│ 
      194│     `initialize_func` is optional, but will be invoked if callable.
      195│     """
    → 196│     for ep in metadata.entry_points(group='keyring.backends'):
      197│         try:
      198│             log.debug('Loading %s', ep.name)
      199│             init_func = ep.load()
      200│             if callable(init_func):

Exited with code exit status 1
```

Upon investigation, I noticed this new complaint from `pip` during installation:

```
ERROR: keyring 23.5.0 has requirement importlib-metadata>=3.6, but you'll have importlib-metadata 1.7.0 which is incompatible.
```

That appears to be a consequence of the change, introduced in poetry 1.1.13, allowing the most recent version of keyring.  In `pyproject.toml` for poetry itself, I see this for v1.1.13:

```
keyring = [
    { version = "^18.0.1", python = "~2.7" },
    { version = "^20.0.1", python = "~3.5" },
    { version = ">=21.2.0", python = "^3.6" }
]
```

and this for 1.1.12:
```
keyring = [
    { version = "^18.0.1", python = "~2.7" },
    { version = "^20.0.1", python = "~3.5" },
    { version = "^21.2.0", python = "^3.6" }
]
```

(note the change from caret to `>=`)

**Comment by clintonroy** [comment]
Pip does the install here, and even warns you about the issue. This appears to be one of the reasons that it's strongly suggested that all python apps are installed in their own virtual environment, to avoid these types of clashes.

**Comment by jclerman** [comment]
@clintonroy: While I understand that pip-installing `poetry` is not the recommended installation technique due to concerns about pollution of the python environment with poetry's own dependencies, that does not seem to be relevant here: There is no other package with which `poetry` has conflicts; instead there is a conflict amongst `poetry`'s own dependencies.

The recommended installation technique is strongly discouraged by our local security policies, and in a CI/CD setting (which is how it's being used when this error comes up), pip-installing poetry seems like it should be OK; very little else is getting installed in the main/default python environment (instead, tox or poetry itself is used to manage dedicated environments).

**Comment by clintonroy** [comment]
Right, gotcha, I think.

If I upgrade pip, it seems to download a different version, which doesn't seem to have the issue you had:

```
FROM docker.io/cimg/python:3.7.10
RUN python3.7 -m venv test
ENV PATH=/home/circleci/project/test/bin/:$PATH
RUN pip install --upgrade pip
RUN pip install poetry
RUN poetry config http-basic.myrepository auser apassword
```

**Comment by camillameri** [comment]
I hit the same issue as jclerman and I have tried with the latest pip (22.0.3).
It looks like substituting `keyring = ">=21.2.0"` (https://github.com/python-poetry/poetry/blob/master/pyproject.toml#L45) with `keyring = ^21.2.0` would fix this?

**Comment by finswimmer** [comment]
Hey,

I'm closing this because it's not a Poetry issue. `pip` seems to wrongly resolve the dependencies here. However we will change the version constraint of `importlib-metadata` to `>=1.6.0` because there is no reason to restrict it more tight.

fin swimmer

**Comment by lucidyan** [comment]
TL;DR solution for those who stuck in the same problem
```
python -m pip install --upgrade pip
```
